### PR TITLE
Update documentation for AMD GPU architecture

### PR DESCRIPTION
**Pull request by Rombur**
Update the documentation for AMD GPU in line with the Nvidia documentation. I have added the documentation about MI300 after getting the approval of AMD.

### CODE REVIEW

**Review by masterleinad — APPROVED** [review]

File: docs/source/keywords.rst [hunk 1]
```diff
@@ -494,77 +494,77 @@ Kokkos will attempt to autodetect the architecture flag at configuration time.
...
+      - Notes
+
+    * * ``Kokkos_ARCH_AMD_GFX942``
+      * GFX942
+      * MI300A, MI300X
+      * (since Kokkos 4.2)
+
+    * * ``Kokkos_ARCH_AMD_GFX940``
+      * GFX940
+      * MI300A (A0 part)
```
Comment (dalg24) [inline docs/source/keywords.rst:521]: Is there a description anywhere of what A0 refers to/mean?

Comment (dalg24) [reply docs/source/keywords.rst:521]: (not asking you to add a link to the page, just wondering how to make sense of that info)

Comment (dalg24) [reply docs/source/keywords.rst:521]: May be we could say "MI300 (pre-production)"

Comment (Rombur) [reply docs/source/keywords.rst]: Done

**Pull request merged**